Build documentation with mkdocs

DavUhll opened the pull request:
Proposal for documentation with `mkdos`.

This PR is shows how to use `mkdocs` to create static documentation which can be later used in CI and host on GitHub pages.

There a quiet some changes in the folder structure, since all the `README.md` are now located in the docs folder.
Maybe not every link is updated correctly.

How to build docs locally:
`pip install -r docs/requirements.txt`
`mkdocs serve`

DavUhll commented:
Let's start fancy stuff with mkdocs @m-naumann 😄

m-naumann commented:
Awesome 🚀 Thanks!

Would you mind adding a CI pipeline for that in your fork, such that we could have a look at the outcome more easily? 😃 

Should be just a few lines, I did that [here](https://github.com/KIT-MRT/behavior_generation_lecture_python/blob/main/.github/workflows/ci.yml#L44-L57) for example. Also, in case people do not like the moving and we don't have too much trouble with links, we could copy the stuff into docs in CI only, as I did there for some stuff.

DavUhll commented:
> Awesome 🚀 Thanks!
> 
> Would you mind adding a CI pipeline for that in your fork, such that we could have a look at the outcome more easily? 😃
> 
> Should be just a few lines, I did that [here](https://github.com/KIT-MRT/behavior_generation_lecture_python/blob/main/.github/workflows/ci.yml#L44-L57) for example. Also, in case people do not like the moving and we don't have too much trouble with links, we could copy the stuff into docs in CI only, as I did there for some stuff.

I've added it.
Could it be, that the CI is only triggered in `Ready for review` PRs?

m-naumann commented:
@DavUhll I added deployment and deployed to https://m-naumann.github.io/Lanelet2-1/, you can merge https://github.com/m-naumann/Lanelet2-1/tree/master into your branch for this update.

Looks nice at the first glance, however some links are broken (hard to fix since they either match the repo or the deployed docs, I guess), and the edit function is broken due to copying the files. But maybe we can go with this for the start. WDYT @immel-f @poggenhans?

immel-f commented:
Looks very nice, thanks :+1: 
I browsed a bit through the deployed version and only found links that point to the repo instead of the website. If that is the only problem, I think we can go with this for now since it is already a big improvement to the existing docs interface

DavUhll commented:
> @DavUhll I added deployment and deployed to https://m-naumann.github.io/Lanelet2-1/, you can merge https://github.com/m-naumann/Lanelet2-1/tree/master into your branch for this update.
> 
> Looks nice at the first glance, however some links are broken (hard to fix since they either match the repo or the deployed docs, I guess), and the edit function is broken due to copying the files. But maybe we can go with this for the start. WDYT @immel-f @poggenhans?

@m-naumann  Thanks for the commit. I've added your commit to this PR.
@immel-f we if you agree I would open this PR and then open a follow up PR to fix all the link to link to the documentation.

let me know what you think about it.

immel-f commented:
> > @DavUhll I added deployment and deployed to https://m-naumann.github.io/Lanelet2-1/, you can merge https://github.com/m-naumann/Lanelet2-1/tree/master into your branch for this update.
> > Looks nice at the first glance, however some links are broken (hard to fix since they either match the repo or the deployed docs, I guess), and the edit function is broken due to copying the files. But maybe we can go with this for the start. WDYT @immel-f @poggenhans?
> 
> @m-naumann Thanks for the commit. I've added your commit to this PR. @immel-f if you agree, I would open this PR and then open a follow up PR to fix all the links to link to the documentation.
> 
> let me know what you think about it.

Sounds great, I am not sure however what to do to fix the links since as I understood it, either the doc or repo links will be broken :sweat_smile: 

What url will the docs be deployed to? We should probably also add a link to the readme before merging.

immel-f commented:
Is this PR ready for merging? I think it would make a very nice addition. If you know to which URL the docs will be deployed to it would be helpful to add a link to the readme as well.

m-naumann commented on README.md:
```diff
@@ -44,6 +44,8 @@ You can find more documentation in the individual packages and in doxygen commen
 - If you are interested in Lanelet2's **projections**, you will find more [here](lanelet2_projection/doc/Map_Projections_Coordinate_Systems.md).
 - To get more information on how to create valid maps, see [here](lanelet2_maps/README.md).
 
+You can also find the documentation at this [link](https://fzi-forschungszentrum-informatik/github.io/lanelet2).
```
```suggestion
You can also find the documentation at this [link](https://fzi-forschungszentrum-informatik.github.io/lanelet2).
```
typo

m-naumann commented:
Looks good to me now: https://m-naumann.github.io/Lanelet2-1/

m-naumann commented on README.md:
```diff
@@ -44,6 +44,8 @@ You can find more documentation in the individual packages and in doxygen commen
 - If you are interested in Lanelet2's **projections**, you will find more [here](lanelet2_projection/doc/Map_Projections_Coordinate_Systems.md).
 - To get more information on how to create valid maps, see [here](lanelet2_maps/README.md).
 
+You can also find the documentation at this [link](https://fzi-forschungszentrum-informatik.github.io/lanelet2).
```
```suggestion
You can also find the documentation at this [link](https://fzi-forschungszentrum-informatik.github.io/Lanelet2).
```

One more typo, appears to be case sensitive

m-naumann commented:
pinging again 😉

immel-f commented:
Thanks for the reminder :sweat_smile:
